lib/coverband/collectors/view_tracker.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'singleton'
+
+module Coverband
+  module Collectors
+    ###
+    # This class tracks view file usage via ActiveSupport::Notifications
+    #
+    # This is a port of Flatfoot, a project I open sourced years ago, but am now rolling into Coverband
+    # https://github.com/livingsocial/flatfoot
+    #
+    # TODO: test and ensure slim, haml, and other support
+    ###
+    class ViewTracker
+      DEFAULT_TARGET = Dir.glob('app/views/**/*.html.erb').reject { |file| file.match(/(_mailer)/) }
+      attr_accessor :target, :logged_views, :views_to_record
+      attr_reader :logger, :roots, :store, :ignore_patterns
+
+      def initialize(options = {})
+        raise NotImplementedError, 'View Tracker requires Rails 4 or greater' unless self.class.supported_version?
+
+        @project_directory = File.expand_path(Coverband.configuration.root)
+        @ignore_patterns = Coverband.configuration.ignore
+        @store = options.fetch(:store) { Coverband.configuration.store }
+        @logger = options.fetch(:logger) { Coverband.configuration.logger }
+        @target = options.fetch(:target) { DEFAULT_TARGET }
+
+        @roots = options.fetch(:roots) { Coverband.configuration.all_root_patterns }
+        @roots = @roots.split(',') if @roots.is_a?(String)
```

Comment:
Feels a little funny that we have to do this in the collector class. Something we might want to clean up later.

Reply:
Yeah, I guess right now this class combines, collector, storage, and reporting concerns. It is nice in that it is small and contains pretty much everything, but does break some of the modularize concerns we had built. 

I think for now it is OK because it is nice to keep it small and compact, and I am not sure as we look at supporting a few different plugin / tools, if the current way we split things up will be flexible enough to support  that.